### DISCUSSION TITLE
How to format swig interface files?

### DISCUSSION THREAD

**Discussion by xuruilong100** [swig/swig#2343] [Q&A] [answered]
How to format swig interface files? Are there some tools?

**Comment by wsfulton** [comment, marked as answer]
The SWIG developers don't use any tools to format the SWIG interface files. Maybe the clang-format tool could be adapted to do so? SWIG interface files are after all just a few SWIG extensions to C/C++.

**Comment by Avasam** [comment]
I also wanted to format `.i` files in `pywin32`. It's limited to my use case for that project (https://github.com/mhammond/pywin32/pull/2746), but I have created, and published, a tool to format SWIG `.i` files using clang-format: https://github.com/Avasam/clang-format-swig
